python/ml4ir/applications/ranking/tests/test_feature_fns.py

```diff
@@ -239,3 +241,53 @@ def test_query_type_vector_without_remove_quotes(self):
         one_hot = np.eye(8)
         expected_one_hot = np.stack([one_hot[i] for i in [0, 1, 2, 3, 4, 5, 6, 7, 5, 5]])
         self.assertTrue(np.isclose(actual_one_hot, expected_one_hot).all())
+
+    def test_query_embedding_vector_using_glove(self):
+        # Setup mock feature_info and file_io
+        feature_info = {
+            "name": "default_feature",
+            "feature_layer_info": {
+                "args": {
+                    "embedding_size": 3,
+                    "glove_path": "mock_glove_path.txt",
+                    "max_entries": 2
+                }
+            }
+        }
+        file_io = MagicMock()
+
+        # Mock the GloVe file content
+        glove_content = "word1 0.1 0.2 0.3\nword2 0.4 0.5 0.6\nword 0.9 0.9 0.9"
+        open_mock = unittest.mock.mock_open(read_data=glove_content)
+        with unittest.mock.patch('builtins.open', open_mock):
+            query_embedding_vector = string_transforms.QueryEmbeddingVector(feature_info, file_io)
+        word1 = tf.constant("word1", dtype=tf.string)
+        word2 = tf.constant("word2", dtype=tf.string)
+
+        # assert dimensions
+        self.assertIn(str(word1), query_embedding_vector.word_vectors)
+        self.assertIn(str(word2), query_embedding_vector.word_vectors)
+        self.assertEqual(len(query_embedding_vector.word_vectors), 2)
+        self.assertEqual(query_embedding_vector.embedding_dim, 3)
+
```

Comment:
@mohazahran these asserts/tests could be separate unit tests or subtests to improve readability.